src/app/system/manage-data-tables/create-data-table/create-data-table.component.ts

```diff
@@ -0,0 +1,175 @@
+   */
+  createDataTableForm() {
+    this.dataTableForm = this.formBuilder.group({
+      'datatableName': ['', Validators.required],
+      'apptableName': ['', Validators.required],
+      'multiRow': [''],
+      'columns': this.formBuilder.array([this.createColumnForm()])
+    });
+  }
+
+  /**
+   * Creates the Column Form.
+   * @return {FormGroup} Column Form.
+   */
+  createColumnForm(): FormGroup {
+    return this.formBuilder.group({
+      'name': ['', Validators.required],
+      'type': ['', Validators.required],
+      'mandatory': [''],
+      'length': [{ value: '', disabled: true }, Validators.required],
+      'code': [{ value: '', disabled: true }, Validators.required]
+    });
+  }
+
+  /**
+   * Gets the columns form array.
+   * @returns {FormArray} Columns form array.
+   */
+  get columns(): FormArray {
+    return this.dataTableForm.get('columns') as FormArray;
+  }
+
+  /**
+   * Adds the column form to given column form array.
+   * @param {FormArray} columnFormArray Given column form array.
+   * Subscribes to the Column Type for the updated Form Array.
+   */
+  addColumn(columnFormArray: FormArray) {
+    columnFormArray.push(this.createColumnForm());
+    this.onColumnTypeChanges();
+  }
+
+  /**
+   * Removes the column form from given column form array at given index.
+   * @param {FormArray} columnFormArray Given column form array.
+   * @param {number} index Array index from where column form needs to be removed.
+   * Subscribes to the Column Type for the updated Form Array.
+   */
+  removeColumn(columnFormArray: FormArray, index: number) {
+    columnFormArray.removeAt(index);
+    this.onColumnTypeChanges();
+  }
+
+  /**
+   * Enables Length Text Field and disables Code Text Field when Column Type is 'String'.
+   * Enables Code Text Field and disables Length Text Field when Column Type is 'Dropdown'.
+   * Disables both Length and Code Text Field when Column Type is neither 'String' nor 'Dropdown'.
+   */
+  onColumnTypeChanges() {
+    const columnFormArray = <FormArray>this.dataTableForm.get('columns');
+    for (let index = 0; index < columnFormArray.length; index++) {
+      columnFormArray.at(index).get('type').valueChanges
+        .subscribe(type => {
+          if (type === 'String') {
+            columnFormArray.at(index).get('length').enable();
+            columnFormArray.at(index).get('code').disable();
+          } else if (type === 'Dropdown') {
+            columnFormArray.at(index).get('code').enable();
+            columnFormArray.at(index).get('length').disable();
+          } else {
+            columnFormArray.at(index).get('code').disable();
+            columnFormArray.at(index).get('length').disable();
+          }
+        });
+    }
+  }
+
+  submit() {
+    const dataTable = this.dataTableForm.value;
+    /**  Checking if any column type is Dropdown and if yes then changing its name to 'code_cd_name' as required by the backend */
+    for (const column of dataTable.columns) {
```

Comment:
Instead of this, change `columnTypeData` to an array of objects with keys `displayValue` and `value` and make all the necessary changes that are required accordingly.